http.go

```diff
@@ -46,11 +46,10 @@ type Request struct {
 
 	keepBodyBuffer bool
 
+	// Used by Server to indicate the request was received on a HTTPS endpoint.
+	// Client/HostClient shouldn't use this field but should depend on the uri.scheme instead.
 	isTLS bool
```

Comment:
Do we really need for this field now?

Reply:
Yeah I also wanted to remove it, but the messy way URIs are handled prevents this. This needs to be set from `Server` so that the `URI.parse` call happening at some point can know if the server was http or https. We can't just set `Request.uri.scheme` instead because a `Request` from a `Server` doesn't use the parsed `uri` but instead has this other form which then later gets parsed. Its all super messy but I don't see how I can improve that all without breaking the `URI/``Request` APIs.